pkg/types/kustomization.go

```diff
@@ -171,6 +172,50 @@ func DealWithDeprecatedFields(data []byte) []byte {
 	return data
 }
 
+func (k *Kustomization) DealWithLegacyKVGenerators() []string {
```

Comment:
this is sufficiently complex that it should have a regression test in kustomization_test.go (which doesn't exist yet).